src/xcode/ENA/ENA/Source/Services/Exposure Transaction/ExposureDetectionExecutor.swift

```diff
@@ -52,7 +54,144 @@ final class ExposureDetectionExecutor: ExposureDetectionDelegate {
 				}
 				group.leave()
 			}
-			#else
+		}
+
+		group.enter()
+
+		client.availableDays(forCountry: country) { result in
+			switch result {
+			case let .success(days):
+				daysAndHours.days = days
+			case let .failure(error):
+				errors.append(error)
+			}
+			group.leave()
+		}
+
+		group.notify(queue: .main) {
+			guard errors.isEmpty else {
+				logError(
+					message: "Unable to determine available data due to errors:\n \(errors.map { $0.localizedDescription }.joined(separator: "\n"))"
+				)
+				completion(/* we are unable to determine the days and hours */ nil, country)
+				return
+			}
+			completion(daysAndHours, country)
+		}
+	}
+
+	func exposureDetection(
+		country: String,
+		downloadDeltaFor remote: DaysAndHours
+	) -> DaysAndHours {
+
+		// prune the store
+		try? downloadedPackagesStore.deleteOutdatedDays(now: .formattedToday())
+
+		let localDays = Set(downloadedPackagesStore.allDays(country: country))
+		let localHours = Set(downloadedPackagesStore.hours(for: .formattedToday(), country: country))
+
+		let delta = DeltaCalculationResult(
+			remoteDays: Set(remote.days),
+			remoteHours: Set(remote.hours),
+			localDays: localDays,
+			localHours: localHours
+		)
+
+		return DaysAndHours(
+			days: Array(delta.missingDays),
+			hours: Array(delta.missingHours)
+		)
+	}
+
+	func exposureDetection(
+		country: String,
+		downloadAndStore delta: DaysAndHours,
+		completion: @escaping (Error?) -> Void
+	) {
+		func storeDaysAndHours(_ fetchedDaysAndHours: FetchedDaysAndHours) {
+			downloadedPackagesStore.addFetchedDaysAndHours(fetchedDaysAndHours, country: country)
+			completion(nil)
+		}
+
+		client.fetchDays(
+				delta.days,
+				hours: delta.hours,
+				of: .formattedToday(),
+				country: country,
+				completion: storeDaysAndHours
+		)
+	}
+
+	func exposureDetectionWriteDownloadedPackages(
+		country: String
+	) -> WrittenPackages? {
+
+		let fileManager = FileManager()
+		let rootDir = fileManager.temporaryDirectory.appendingPathComponent(UUID().uuidString, isDirectory: true)
+		do {
+			try fileManager.createDirectory(at: rootDir, withIntermediateDirectories: true, attributes: nil)
+
+			let writer = AppleFilesWriter(rootDir: rootDir)
+
+			if store.hourlyFetchingEnabled {
+				let hourlyPackages = downloadedPackagesStore.hourlyPackages(for: .formattedToday(), country: country)
```

Comment:
Not sure if the following aspect is handled somewhere else but to us `hourlyFetchingEnabled` means that we only use the **three** most recent hours - not all hours from today. I was unable to find this logic in our current *develop* but it was there up until recently. Do you know what happened there? :) 

<img width="654" alt="Screen Shot 2020-09-11 at 18 26 09" src="https://user-images.githubusercontent.com/153225/92950106-7082a480-f45c-11ea-8bf6-92892758c4aa.png">

Reply:
Good one. I think it got lost on the way. Willl add it again